activesupport/test/core_ext/marshal_test.rb

```diff
@@ -19,6 +19,25 @@ def teardown
     end
   end
 
+  test "that Marshal#load still works when passed a proc" do
+    example_string = "test"
+
+    example_proc = Proc.new do |o|
+      if o.is_a?(String)
+        o.capitalize!
+      end
+    end
+
+    dumped = Marshal.dump(example_string)
+
+    assert_nothing_raised do
+      Marshal.load(dumped, example_proc)
+    end
+
+    assert_equal Marshal.load(dumped, example_proc), "Test"
+    assert_equal Marshal.load(dumped, example_proc), Marshal.method(:load).super_method.call(dumped, example_proc)
```

Comment:
Thanks for your feedback. I guess we don't need lines 33-35, because by testing the output of Marshal#load we are also testing that the function doesn't raise an error. And I could see why line 38 is arguably testing the Marshal ruby class, not the ActiveSupport extension.